### PR TITLE
Terraform domain configuration

### PR DESCRIPTION
**Pull request by smneal**
## Context
Currently the DNS for the Apply service is managed in AWS Route 53 and the process of updating records is quite a manual one which requires various requests to be raised.

This change will create the Azure resources required to managed the DNS using Azure DNS. It will improve the traceability of changes, and lead to the ability to quickly make changes to the services DNS records as required.

## Changes proposed in this pull request

Add the Terraform code to manage the services domains using Azure DNS.

## Link to Trello card

https://trello.com/c/1yEerHJZ/71-migrate-apply-dns-zones-to-azure-dns

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: dns/readme.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    - dns/zones/workspace_variables/backend_${zone}.tfvars
+
+- Add the zone to the Makefile
+
+- Run the make command (there is no workflow job for zone creation or update)
+
+    - `make ${zone} dnszone-plan`
+    - `make ${zone} dnszone-apply`
+
+- Provide the NS records for delegation from service.gov.uk
```
Comment (saliceti) [inline dns/readme.md:35]: or from education.gov.uk

**Review by saliceti — APPROVED** [review]
👏

File: dns/readme.md (hunk 1 above)
Comment (smneal) [reply dns/readme.md]: Done

**Review by saliceti — APPROVED** [review]

File: dns/zones/workspace-variables/apply-zone.tfvars.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+          "value": "v=spf1 -all"
+        },
+        "_dmarc": {
+          "value": "v=DMARC1; p=reject; sp=reject; rua=mailto:dmarc-rua@dmarc.service.gov.uk; ruf=mailto:dmarc-ruf@dmarc.service.gov.uk"
+        }
+      },
+      "resource_group_name": "s189p01-applydomains-rg"
+    }
+  }
+}
```
Comment (RMcVelia) [inline dns/zones/workspace-variables/apply-zone.tfvars.json:17]: Missing CAA record?

Comment (smneal) [reply dns/zones/workspace-variables/apply-zone.tfvars.json:17]: Yes - we decided we couldn't add one of these in just yet.

**Pull request merged**